File: VocaDbModel/DataContracts/Artists/CreateArtistContract.cs
#nullable disable

using VocaDb.Model.Domain.Artists;

namespace VocaDb.Model.DataContracts.Artists
{
	public class CreateArtistContract
	{
		public ArtistType ArtistType { get; init; }

		public string Description { get; init; }

		public bool Draft { get; init; }

		public LocalizedStringContract[] Names { get; init; }

		public EntryPictureFileContract PictureData { get; set; }

		public WebLinkContract WebLink { get; init; }
	}
}

File: VocaDbWeb/Scripts/DataContracts/Artist/CreateArtistContract.ts
import ArtistType from '@/Models/Artists/ArtistType';

import EntryPictureFileContract from '../EntryPictureFileContract';
import LocalizedStringContract from '../Globalization/LocalizedStringContract';
import WebLinkContract from '../WebLinkContract';

// Corresponds to the CreateArtistForApiContract record class in C#.
export default interface CreateArtistContract {
	artistType: ArtistType;
	description: string;
	draft: boolean;
	names: LocalizedStringContract[];
	pictureData?: EntryPictureFileContract;
	webLink?: WebLinkContract;
}
